src/glue/jobs/s3_to_json.py

```diff
@@ -0,0 +1,209 @@
+                j["Type"] = metadata["subtype"]
+            data.append(j)
+    if dataset_identifier == "HealthKitV2Samples":
+        output_fname = "{}_{}_{}-{}.ndjson".format(
+                metadata["type"],
+                metadata["subtype"],
+                metadata["start_date"].strftime("%Y%m%d"),
+                metadata["end_date"].strftime("%Y%m%d")
+        )
+    elif metadata["start_date"] is None:
+        output_fname = "{}_{}.ndjson".format(
+                metadata["type"],
+                metadata["end_date"].strftime("%Y%m%d")
+    )
+    else:
+        output_fname = "{}_{}-{}.ndjson".format(
+                metadata["type"],
+                metadata["start_date"].strftime("%Y%m%d"),
+                metadata["end_date"].strftime("%Y%m%d")
+        )
+    output_path = os.path.join(dataset_identifier, output_fname)
+    s3_output_key = os.path.join(
+        workflow_run_properties["json_prefix"],
+        f"dataset={dataset_identifier}",
+        output_fname)
+    logger.debug(f"Output Key: {s3_output_key}")
+    with open(output_path, "w") as f_out:
+        for record in data:
+            f_out.write("{}\n".format(json.dumps(record)))
+    with open(output_path, "rb") as f_in:
+        response = s3_client.put_object(
+                Body = f_in,
+                Bucket = workflow_run_properties["json_bucket"],
+                Key = s3_output_key,
+                Metadata = s3_metadata)
+        logger.debug(f"S3 Put object response: {json.dumps(response)}")
+    return output_path
+
+def get_metadata(basename):
+    """
+    Get metadata of a file by parsing its basename.
+
+    Args:
+        basename (str): The basename of the file.
+
+    Returns:
+        dict: The metadata, formatted as
+            type (str): the data type
+            start_date (datetime.datetime): The date of the oldest data collected.
+                May be None if filename only contains the `end_date`.
+            end_date (datetime.datetime): The date of the most recent data collected.
+            subtype (str): If this is HealthKitV2Samples, the type of the
+                sample data. (Not to be confused with the data type, which in
+                this case is HealthKitV2Samples or HealthKitV2Samples_Deleted)
+    """
+    logger = logging.getLogger(__name__)
+    metadata = {}
+    basename_components = os.path.splitext(basename)[0].split("_")
+    metadata["type"] = basename_components[0]
+    if "-" in basename_components[-1]:
+        start_date, end_date = basename_components[-1].split("-")
+        metadata["start_date"] = \
+                datetime.datetime.strptime(start_date, "%Y%m%d")
+        metadata["end_date"] = \
+                datetime.datetime.strptime(end_date, "%Y%m%d")
+    else:
+        metadata["start_date"] = None
+        metadata["end_date"] = \
+                datetime.datetime.strptime(basename_components[-1], "%Y%m%d")
```

Comment:
Could you provide an example of what a basename looks like?

Reply:
See the page I linked you to above: https://sagebionetworks.jira.com/wiki/spaces/IBC/pages/2789376001/Notes#Types-of-data-included-in-archive

For example, `HealthKitV2Samples_ActiveEnergyBurned_20220111-20230103.json`